plugin/trino-clickhouse/src/main/java/io/trino/plugin/clickhouse/ClickHouseTableProperties.java

```diff
@@ -36,26 +42,120 @@
         implements TablePropertiesProvider
 {
     public static final String ENGINE_PROPERTY = "engine";
-    public static final String DEFAULT_TABLE_ENGINE = "Log";
+    // MergeTree engine properties
+    public static final String ORDER_BY_PROPERTY = "order_by"; //required
+    public static final String PARTITION_BY_PROPERTY = "partition_by"; //optional
+    public static final String PRIMARY_KEY_PROPERTY = "primary_key"; //optional
+    public static final String SAMPLE_BY_PROPERTY = "sample_by"; //optional
+
+    public static final ClickHouseEngineType DEFAULT_TABLE_ENGINE = ClickHouseEngineType.LOG;
 
     private final List<PropertyMetadata<?>> tableProperties;
 
     @Inject
     public ClickHouseTableProperties()
     {
         tableProperties = ImmutableList.of(
-                stringProperty(
+                enumProperty(
                         ENGINE_PROPERTY,
                         "ClickHouse Table Engine, defaults to Log",
+                        ClickHouseEngineType.class,
                         DEFAULT_TABLE_ENGINE,
+                        false),
+                new PropertyMetadata<>(
+                        ORDER_BY_PROPERTY,
+                        "columns to be the sorting key. Required",
```

Comment:
"Required for tables using MergeTree engine property."

But I think it's OK to drop "Required" and "Optional" here and just describe this in docs.

Reply:
These description of table property comes from office documentation.
Ok, I change them.